### ISSUE TITLE
Collapsed blocks

### ISSUE THREAD

**Issue by hanche** [heyman/heynote#54]
For collapsed blocks, I think it is better to show the first line rather than the last line.
The first line is more likely to give a clue what the block is about.

It might even be nice to be able to give a separate title to blocks and have collapsed blocks showing only the title, but now I am possibly courting creeping featuritis.

However, I would very much like keybindings to collaps/expand the current block. For example (on the Mac) ⌘ + Left to collapse, ⌘ + Right to expand. Or perhaps one of those to toggle the collapsed state.

**Comment by heyman** [comment]
> For collapsed blocks, I think it is better to show the first line rather than the last line.

I agree. Most of the current collapsing of block comes from CodeMirror code folding functionality. I remember tinkering with it awhile ago, but IIRC I kept running onto some issue (but I can't exactly remember what the problem was). 

> It might even be nice to be able to give a separate title to blocks

I don't want to add named blocks, since I think it goes against the "scratchpadness".

> However, I would very much like keybindings to collaps/expand the current block.

Yes, sounds like a good idea.

**Comment by simonratner** [comment]
> > It might even be nice to be able to give a separate title to blocks
>
> I don't want to add named blocks, since I think it goes against the "scratchpadness".

Agree but i think a lot can be done with existing info without adding "named blocks". e.g. a collapsed status line like this:
```
‣ First line (truncated...) [Markdown] [4 lines]
```

Especially for code blocks, first line will often be a comment or a declaration which provides a nice summary of the block already.
For comparison, this is how collapsed blocks look in my vim setup: 
![image](https://github.com/heyman/heynote/assets/667283/0eecb96b-5af5-483c-b19d-31187504bd19)
